storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java

```diff
@@ -0,0 +1,355 @@
+            return;
+        }
+        if (existsNode(zk, npath, false)) {
+            return;
+        }
+        byte[] byteArray = new byte[1];
+        byteArray[0] = (byte) 7;
+        createNode(zk, npath, byteArray, org.apache.zookeeper.CreateMode.PERSISTENT, acls);
+
+    }
+
+    public static void syncPath(CuratorFramework zk, String path) throws RuntimeException {
+        try {
+            zk.sync().forPath(Utils.normalizePath(path));
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    public static void addListener(CuratorFramework zk, ConnectionStateListener listener) {
+        zk.getConnectionStateListenable().addListener(listener);
+    }
+
+    public static byte[] getData(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            if (existsNode(zk, npath, watch)) {
+                if (watch) {
+                    return zk.getData().watched().forPath(npath);
+                } else {
+                    return zk.getData().forPath(npath);
+                }
+            }
+        } catch (KeeperException e) {
+            // this is fine b/c we still have a watch from the successful exists call
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        return null;
+    }
+
+    public static Integer getVersion(CuratorFramework zk, String path, boolean watch) throws Exception {
+        String npath = Utils.normalizePath(path);
+        Stat stat = null;
+        if (existsNode(zk, npath, watch)) {
+            if (watch) {
+                stat = zk.checkExists().watched().forPath(Utils.normalizePath(path));
+            } else {
+                stat = zk.checkExists().forPath(Utils.normalizePath(path));
+            }
+            return Integer.valueOf(stat.getVersion());
+        }
+
+        return null;
+    }
+
+    public static List<String> getChildren(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+
+        try {
+            String npath = Utils.normalizePath(path);
+            if (watch) {
+                return zk.getChildren().watched().forPath(npath);
+            } else {
+                return zk.getChildren().forPath(npath);
+            }
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+    }
+
+    // Deletes the state inside the zookeeper for a key, for which the
+    // contents of the key starts with nimbus host port information
+    public static void deleteDodeBlobstore(CuratorFramework zk, String parentPath, String hostPortInfo) throws RuntimeException {
+        String parentnPath = Utils.normalizePath(parentPath);
+        List<String> childPathList = null;
+        if (existsNode(zk, parentnPath, false)) {
+            childPathList = getChildren(zk, parentnPath, false);
+        }
+        for (String child : childPathList) {
```

Comment:
If the parent path does not exist we will get an NPE here that we would not get before.  Please move this under the existsNode check.